lib/Sema/ConstraintSystem.cpp

```diff
@@ -712,30 +712,54 @@ void ConstraintSystem::recordOpenedTypes(
 
 /// Determine how many levels of argument labels should be removed from the
 /// function type when referencing the given declaration.
-static unsigned getNumRemovedArgumentLabels(ASTContext &ctx, ValueDecl *decl,
+static unsigned getNumRemovedArgumentLabels(TypeChecker &TC, ValueDecl *decl,
                                             bool isCurriedInstanceReference,
                                             FunctionRefKind functionRefKind) {
+  unsigned numParameterLists = 0;
+
+  // This is required because if enum case has a function type and passed
```

Comment:
Please either remove this comment, or rewrite it to talk about the common behavior of both functions and enum cases.